Documentation/Articles/querying_for_data.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Querying for Data
+description: About Reading data
+keywords: read model, query, ddd, read, cqrs,  
+author: pavsaund
+---
+
+
+# Querying for Data
+
+## Introduction
+The query or read side of the CQRS pattern is all about exposing the current state of a system in a format that is ready to be consumed by interaction layers (ie: web-based user interface). The state should be exposed through models that are optimized for querying, and is usually updated by changes made to a system through events.
+
+{{% notice note %}}
+Dolittle exposes read data through two different mechanisms:
+ 
+ - [Read Models](https://github.com/dolittle/Runtime/tree/master/Documentation/read/read_model.md)
+ - [Queries](https://github.com/dolittle/Runtime/tree/master/Documentation/read/query.md)
+{{% /notice %}}
+
+### Updating Read Data
+[Event Processors](LINK TO EVENT PROCESSORS) are responsible for consuming events in the system and updating read-models.
+
+### Storing Data 
+#### Optimize for querying
+Unlike traditional relational data models that are spread across tables, read data should be optimized as much as possible to the clients consuming that data. This means creating models that can be consumed directly without additional queries. One of the of the benefits is that reading data becomes lightweight, fast and allows for good user experiences.
+
+Duplicating data in 
+
+{{% notice tip %}}
+Think of data on the query side as your applications cache. It get's updated whenever there's a change to the system, and is fast by default.
+{{% /notice %}}
+
+##### Example of optimizing for querying:
+In an e-commerce solution, you get a requirement to show a small-cart summary with Number of Items and total price. This is a challenge since it will get the current ShoppingCartModel each time the user hits any page of the site, and it adds a visible delay to the user.
```

Comment:
I'd emphasise having a fit-for-purpose model over the full ShoppingCartModel being slow (as it shouldn't be when it's pre-calculated, excepting bandwidth considerations). The point is to have a model that slots nicely into the need of the view.